test/integration/services/alert.js

```diff
@@ -0,0 +1,89 @@
+describe('Alert Service', function() {
+  specify('Displaying', function() {
+    cy
+      .visit('/');
+
+    cy
+      .clock();
+
+    cy
+      .window()
+      .should('have.property', 'Radio')
+      .then(Radio => {
+        Radio.request('alert', 'show:success', 'success');
+      });
+
+    cy
+      .get('.alert-box')
+      .contains('success')
+      .should('exist')
+      .tick(4900);
+
+    cy
+      .get('.alert-box')
+      .should('not.exist');
+
+    cy
+      .clock()
+      .then((clock) => {
+        clock.restore();
+      });
+
+    cy
+      .window()
+      .should('have.property', 'Radio')
+      .then(Radio => {
+        Radio.request('alert', 'show:info', 'info');
+      });
+
+    cy
+      .get('.alert-box')
+      .contains('info');
+
+    cy
+      .window()
+      .should('have.property', 'Radio')
+      .then(Radio => {
+        Radio.request('alert', 'show:error', 'error');
+      });
+
+    cy
+      .get('.alert-box')
+      .contains('error');
+
+    const options = {
+      onUndo: cy.stub(),
+      onComplete: cy.stub(),
+    };
+
+    cy
+      .window()
+      .should('have.property', 'Radio')
+      .then(Radio => {
+        Radio.request('alert', 'show:undo', options);
+      });
+
+    cy
+      .get('.alert-box')
+      .find('.js-dismiss')
+      .click()
+      .then(() => {
+        expect(options.onComplete).to.be.called;
```

Comment:
let's be specific with `calledOnce`

Reply:
So looks like there might be an issue with the undo then 🤔. The stub is being called twice...I'll keep looking, but any extra eyes you wanna throw on there would be helpful

Reply:
sure enough.   We need to do `    const alertView = this.showAlert(_.omit(options, 'onUndo', 'onComplete'));` in the alert service.  `triggerMethod` currently looks at passed in options as possible handlers, so it's duplicating the handler.  This is due to be removed in a future version of Mn, so we should pull out the handlers so that they're not passed to the view for now.